service/topology_coordinator.cc

```diff
@@ -822,6 +825,58 @@ class topology_coordinator : public endpoint_lifecycle_subscriber {
         }
     }
 
+    future<> group0_voter_refresher_fiber() {
+        rtlogger.debug("start group0 members refresh fiber");
+        while (!_as.abort_requested()) {
+            bool sleep = false;
+            try {
+                co_await _topo_sm.event.when([&] {
+                    return !_group0_voter_events.empty() || _as.abort_requested();
+                });
+                const auto& ep_to_id = co_await _sys_ks.load_host_ids();
+
+                const auto events = _group0_voter_events | std::ranges::views::transform([&ep_to_id](const auto& event) {
+                    const auto& [endpoint, ep_can_vote] = event;
+                    auto it = ep_to_id.find(endpoint);
+                    if (it == ep_to_id.end()) {
+                        rtlogger.warn("group0 members refresh fiber: no host_id found for endpoint {}", endpoint);
+                    }
+                    return std::make_tuple(it, ep_can_vote);
+                }) | std::ranges::views::filter([&ep_to_id](const auto& event) {
+                    const auto& [it, ep_can_vote] = event;
+                    return it != ep_to_id.end();
+                }) | std::ranges::views::transform([](const auto& event) {
+                    const auto& [it, ep_can_vote] = event;
+                    return std::make_tuple(raft::server_id(it->second.uuid()), ep_can_vote);
+                }) | std::ranges::to<std::vector>();
+
+                _group0_voter_events.clear();
+
+                for (const auto& [id, ep_can_vote] : events) {
+                    co_await _group0.set_voter_status(id, ep_can_vote, _as);
+                }
+            } catch (raft::request_aborted&) {
+                rtlogger.debug("group0 voters refresh fiber aborted");
+            } catch (seastar::abort_requested_exception) {
+                rtlogger.debug("group0 voters refresh fiber aborted");
+            } catch (group0_concurrent_modification&) {
```

Comment:
So if this is thrown but `_group0_voter_events` is already cleared we lose events. This shows that we need to handle this like level triggering event and not edge triggering.